### PR TITLE
Documentation: Supervisor, worker, and create a worker internal guide

### PR DESCRIPTION
**Pull request by EdenComp**
Also fixes ARE-245
Also fixes ARE-247

### CODE REVIEW

File: docs/architecture/archi.png
Comment (RezaRahemtola) [inline docs/architecture/archi.png:1]: move to the assets folder

File: docs/architecture/supervisor.md [hunk 1]
```diff
@@ -1 +1,71 @@
...
+
+The supervisor role is to orchestrate the execution of jobs. It receives gRPC requests from the backend and executes the jobs with the passed parameters.
+
+The job parameters are passed using the command arguments of the container.
+
+Workers communicate directly with the backend to send data and errors.
+
+## Requirements 
+
+The supervisor runs in a **Docker container** and is connected to the **Docker Engine API**, allowing it to create and remove containers.
```
Comment (RezaRahemtola) [inline docs/architecture/supervisor.md:18]: maybe add link to docker engine API

File: docs/architecture/supervisor.md [hunk 2]
```diff
@@ -1 +1,71 @@
...
+
+## Requests and responses
+
+The supervisor communicates with the backend using gRPC. Declarations and prototypes are available on the side repository [area-proto](https://github.com/RezaRahemtola/area-proto)
+
+The supervisor responds to the following requests:
+
+### LaunchJob: Launch a job with the given parameters
+
+The job parameters must have been checked before sending to the supervisor.<br/>
```
Comment (RezaRahemtola) [inline docs/architecture/supervisor.md:29]: you can use `\` instead of `<br/>` but doesn't matter

File: docs/architecture/supervisor.md [hunk 3]
```diff
@@ -1 +1,71 @@
...
+The supervisor will stop the job with the given identifier.<br/>
+If the job is not running, the supervisor won't do anything and return successfully.<br/>
+
+### KillAllJobs: Stop all running jobs
+
+The supervisor will stop all running jobs.<br/>
+
+## Debug
+
+The ``GO_ENV`` variable can be used to enable development or production mode.
```
Comment (RezaRahemtola) [inline docs/architecture/supervisor.md:48]:
```suggestion
The `GO_ENV` variable can be used to enable development or production mode.
```

File: docs/internal/create-workers.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+If an error occurs, you must send it to the backend using the [OnError](../architecture/workers.md#onerror) gRPC route.
+
+### Dockerisation
+
+The last step of the creation of a *worker* is to create a Dockerfile to build it.
+
+The Dockerfile must be located in the root of the worker directory.
+
+You also have to register your worker in the [Supervisor docker-compose.yaml](../../backend/supervisor/docker-compose.yml).
```
Comment (RezaRahemtola) [inline docs/internal/create-workers.md:40]: this link won't work in prod, add a full link with github.com

File: docs/internal/create-workers.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+To register a job, please add it to the following files:
+
+- The jobs list in the backend: [jobs.ts](../../backend/back/src/types/jobs.ts)
+> You can create a template class that will be automatically validated in the [jobParams.ts file](../../backend/back/src/types/jobParams.ts)
+
+- The jobs identifier factory: [jobIds.ts](../../backend/back/src/types/jobIds.ts)
+> :warn: The job identifier factory is used to generate the unique identifier of a job. It must be unique for every different job. (See [Jobs identification](../architecture/supervisor.md#jobs-identification))
+
+- The [supervisor data](../../backend/supervisor/jobs/images.go) in the supervisor
```
Comment (RezaRahemtola) [inline docs/internal/create-workers.md:56]: same for the links

File: docs/internal/create-workers.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+    {
+        name: string
+        type: string
+        required: boolean,
+        values?: string[]
+    }
+]
+```
+
+> :bulb: The supported types are available [here](../../backend/back/src/services/dto/area.dto.ts)
```
Comment (RezaRahemtola) [inline docs/internal/create-workers.md:104]: same

File: docs/internal/create-workers.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+
+> The `values` property is only needed for the `select` type.
+
+#### Parameters return flow
+
+The parameters return flow is the list of the keys your job will return in the `data` property of the [OnAction / OnReaction gRPC routes](../architecture/workers.md#onaction--onreaction).
```
Comment (RezaRahemtola) [inline docs/internal/create-workers.md:110]: not sure this link works

**Review by RezaRahemtola — CHANGES_REQUESTED** [review]

File: docs/internal/create-workers.md (hunk 7 above)
Comment (EdenComp) [reply docs/internal/create-workers.md:110]: Works on the IDE + github

**Pull request merged**